pkg/apps/pgs/uploader.go

```diff
@@ -551,12 +556,24 @@ func (h *UploadAssetHandler) writeAsset(s *pssh.SSHServerConnSession, reader io.
 		"filename", assetFilepath,
 	)
 
-	_, fsize, err := h.Cfg.Storage.PutObject(
-		data.Bucket,
-		assetFilepath,
-		reader,
-		data.FileEntry,
-	)
+	var fsize int64
+	var err error
+
+	if reader == nil {
+		_, err = h.Cfg.Storage.PutEmptyObject(
+			data.Bucket,
+			assetFilepath,
+			data.FileEntry,
+		)
```

Comment:
Why do we need this if the reader is nil?

Reply:
Reader was never `nil` previously. Before, we'd send an empty reader. Instead of reading it to figure out if we're empty, we can just pass nil now and it'll use the right version of PutObject from the minio client. This has to do with garage not liking multi-part uploads when the content is empty: https://git.deuxfleurs.fr/Deuxfleurs/garage/issues/759